src/interpreter/lexical-environment/get-dot-path-from-node.ts

```diff
@@ -21,8 +21,8 @@ export function getDotPathFromNode<T extends TS.Node>(options: EvaluatorOptions<
 		return SUPER_SYMBOL;
 	} else if (typescript.isParenthesizedExpression(node)) {
 		return getDotPathFromNode({...options, node: node.expression});
-	} else if (typescript.isTypeAssertionExpression?.(node) || (!("isTypeAssertionExpression" in typescript) && (typescript as typeof TS).isTypeAssertion(node))) {
-		return getDotPathFromNode({...options, node: node.expression});
+	} else if (typescript.isTypeAssertionExpression?.(node) || (!("isTypeAssertionExpression" in typescript) && (typescript as any).isTypeAssertion(node))) {
+		return getDotPathFromNode({...options, node: (node as any as TS.TypeAssertion).expression});
```

Comment:
...Then we won't have to do the cast here and can't just keep it as `node.expression`, as TypeScript will understand that `node` is a `TypeAssertionExpression`.